### DISCUSSION TITLE
How to stream private video using c#

### DISCUSSION THREAD

**Discussion by chris-herring** [supabase/supabase#30486] [Questions]
I have a c# backend and a javascript (React) front end. Sign in to Supabase is done on the backend. How do I stream a private video on the client?
I'd rather not use a signed url as that requires me to set a time limit to be set which may not sync with session times plus the url can be shared. 
Is there a way to share the access code from sign in with the javascript library? I could use the password to sign in on the client as well but that won't work if 2FA is used. I looked at grabbing the Access Token from the server side login and using setAuth on the client but that's depreciated.

**Comment by GaryAustin1** [comment]
You can use the access token in the authorization header of createClient to act as that user.  You will not have a true auth session, but all the API calls will work.  You may need to deal with the case of it expiring though.

**Reply by chris-herring** [reply]
Thanks, this did work to make API calls however I had not realised that videos in private buckets can only be streamed by creating a signed url. So I don't think this suits my use case.
